ex/high_throughput_inference/mock_app_redis.py

```diff
@@ -73,16 +77,18 @@ def name(self):
             timing = [batch_size]
             logger.info(f"Iteration: {iteration_number}")
             start = time.perf_counter()
-            client.put_tensor(name="batch", data=resnet.get_batch(batch_size).numpy())
-            client.run_model(name=resnet.name, inputs=["batch"], outputs=["result"])
-            result = client.get_tensor(name="result")
+            input_name = f"batch_{rank}"
+            output_name = f"result_{rank}"
+            client.put_tensor(name=input_name, data=resnet.get_batch(batch_size).numpy())
+            client.run_model(name=resnet.name, inputs=[input_name], outputs=[output_name])
+            result = client.get_tensor(name=output_name)
             end = time.perf_counter()
             timing.append(end-start)
             timings.append(timing)
 
 
 
     timings_np = numpy.asarray(timings)
-    numpy.save("timings.npy", timings_np)
+    numpy.save(f"timings_{rank}.npy", timings_np)
```

Comment:
the perf timers currently have file name information supplied in their constructor. should we use that instead of having another, disconnected filename here?

Reply:
could we have `def persist(self) -> None` on the timer that saves the file (if not using a logger) to avoid having to re-implement combining & saving timer data in each client?